jobs/periodic/flag-old-conversations/index.ts

```diff
@@ -55,42 +55,58 @@ export default async function flagInactiveConversationsAsReadonly() {
         const { readMembers } = checkChatMembersAccessRights(conversation);
         const isChatReadOnly = readMembers.length === countParticipants;
         if (isChatReadOnly) {
-            return;
+            logger.info(`Conversation ${conversation.id} is already readonly.`);
         }
 
-        if (conversation.custom.subcourse) {
-            const subcourseIds: number[] = JSON.parse(conversation.custom.subcourse);
-            const allSubcoursesActive = await Promise.all(subcourseIds.map((id) => isActiveSubcourse(id)));
-            shouldMarkAsReadonly = allSubcoursesActive.every((active) => active === false);
-        }
-        if (conversation.custom.prospectSubcourse) {
-            const prospectSubcourses: number[] = conversation.custom.prospectSubcourse;
-            const allProspectSubcoursesActive = await Promise.all(prospectSubcourses.map((id) => isActiveSubcourse(id)));
-            shouldMarkAsReadonly = allProspectSubcoursesActive.every((active) => active === false);
-        }
+        if (!isChatReadOnly) {
+            if (conversation.custom.subcourse) {
+                const subcourseIds: number[] = JSON.parse(conversation.custom.subcourse);
+                const allSubcoursesActive = await Promise.all(subcourseIds.map((id) => isActiveSubcourse(id)));
+                shouldMarkAsReadonly = allSubcoursesActive.every((active) => active === false);
+            }
+            if (conversation.custom.prospectSubcourse) {
+                const prospectSubcourses: number[] = conversation.custom.prospectSubcourse;
+                const allProspectSubcoursesActive = await Promise.all(prospectSubcourses.map((id) => isActiveSubcourse(id)));
+                shouldMarkAsReadonly = allProspectSubcoursesActive.every((active) => active === false);
```

Comment:
This potentially overrides shouldMarkAsReadonly=false with true

Reply:
why? Any other suggestions?

Reply:
shouldMarkAsReadonly = shouldMarkAsReadonly && ...